### PR TITLE
[MISC] Alignment: aminoacid scoring scheme - lower case member constants

### PR DESCRIPTION
**Pull request by Irallia**
Resolves https://github.com/seqan/product_backlog/issues/317

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -116,6 +116,15 @@ You can either directly check the reported code or verify with our documentation
 For a complete list of behavioural changes in our public and internal API, you can consult our API stability
 regression test suite and patches at https://github.com/seqan/seqan3/tree/master/test/api_stablility/3.0.2/.
 
+#### Alignment
+
+* The member constants of `seqan3::aminoacid_similarity_matrix` were changed to lower case
+  ([\#....](https://github.com/seqan/seqan3/pull/....)):
```
Comment (Irallia) [inline CHANGELOG.md:122]:
```suggestion
  ([\#2599](https://github.com/seqan/seqan3/pull/2599)):
```

File: test/api_stability/3.0.2/0002-NOAPI-DETAIL-MISC-Remove-top-level-scoring-scheme-de.patch [hunk 2]
```diff
@@ -20,7 +20,7 @@ index ef141a7e9..7b0927215 100644
 -                                                                seqan3::align_cfg::method_global,
 -                                                                seqan3::aminoacid_scoring_scheme<>>;
 +                                                                seqan3::align_cfg::method_global>;
- 
+
```
Comment (SGSSGene) [inline test/api_stability/3.0.2/0002-NOAPI-DETAIL-MISC-Remove-top-level-scoring-scheme-de.patch:23]: This doesn't look necessary for a BLOSUM80 -> blosum80 change.
Did this accidentally get into the commit? Or is it something that was forgotten before?

**Review by SGSSGene — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Irallia) [reply CHANGELOG.md:122]: Yes, I marked it for me to make the change when I received your review. Sorry for the confusion.

File: include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp [hunk 3]
```diff
@@ -28,33 +28,39 @@ namespace seqan3
  *
  * This enum provides IDs for amino acid similarity matrixes of the
  * [BLOSUM](https://en.wikipedia.org/wiki/BLOSUM) and [PAM](https://en.wikipedia.org/wiki/Point_accepted_mutation)
- * families.
+ * families (last access 04.05.2021).
```
[the remark this answers is not in the document]
Comment (eseiler) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp:31]: I'm not a big fan of this (in this context, having a bibliography is something else :)):

* No one will ever update the dates even if they check the links
* We don't use any information from the links, it's just some `\seealso` basically
* Access dates are implicitly stored in the git history

File: CHANGELOG.md (hunk 1 above)
Comment (eseiler) [reply CHANGELOG.md:122]: I guess it's for when other changes are applied

File: include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (marehr) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp:31]: Agree with @eseiler; if we can somehow automatically add that date I would love that, but manual management isn't feasible.

[the remark this answers is not in the document]
Comment (Irallia) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp:31]: I'm okay with undoing it. I just want to point out that I had discussed and started this in another PR with @smehringer  and @MitraDarja.
-> https://github.com/seqan/seqan3/pull/2290#discussion_r535200262

Why do you think you do not update this date when you edit the documentation for it?
And wouldn't an automatic check be error-prone? Then one checks only whether the page is there, but not whether it still contains what we have taken over or to which we want to refer?

[the remark this answers is not in the document]
Comment (SGSSGene) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp]: The "last access" only has to be updated if the enum provided IDs on wikipedia change. Isn't this very helpful? One could just look at the wikipedia diff from 04.05.2021 until today and see if anything has changed.

[the remark this answers is not in the document]
Comment (eseiler) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp]: > The "last access" only has to be updated if the enum provided IDs on wikipedia change. Isn't this very helpful? One could just look at the wikipedia diff from 04.05.2021 until today and see if anything has changed.

But the IDs come from some paper, it's not an invention of Wikipedia. The papers won't change and to check for some change would require you to manually look into it. I don't see this happen.

If it's not automatic, it's not reliable (include order, first `brief` or `author`, `|` at end or beginning of lines, how are lambdas intended, are the used URLs at all valid?).

> Why do you think you do not update this date when you edit the documentation for it?

Easy to overlook if I don't even touch text in vicinity of the URL; then you would rely on someone else to know that there are links and verify them. And we have quite a few of them....

[the remark this answers is not in the document]
Comment (Irallia) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp]: I understand the objection, but I would trust us that one gets into the habit of looking for such links. But I have no strong feelings and have undone it.

[the remark this answers is not in the document]
Comment (SGSSGene) [reply include/seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp]: So if I understand correctly, your objection is actually against citing Wikipedia and rather cite the original paper?

**Review by SGSSGene — APPROVED** [review]

File: test/api_stability/3.0.2/0002-NOAPI-DETAIL-MISC-Remove-top-level-scoring-scheme-de.patch [hunk 4]
```diff
@@ -20,7 +20,7 @@ index ef141a7e9..7b0927215 100644
 -                                                                seqan3::align_cfg::method_global,
```
Comment (marehr) [inline test/api_stability/3.0.2/0002-NOAPI-DETAIL-MISC-Remove-top-level-scoring-scheme-de.patch:20]: please don't change patches, if they don't work they need to be changed differently :)

**Review by marehr — COMMENTED** [review]
nearly done, api stability patches shouldn't be changed :)

**Review by marehr — APPROVED** [review]

**Pull request merged**